contracts/HATVaults.sol

```diff
@@ -949,20 +939,48 @@ contract  HATVaults is Governable, ReentrancyGuard {
                     bool _depositPause,
                     string memory _descriptionHash)
     external onlyGovernance {
-        require(poolInfos.length > _pid, "HVE23");
+        require(poolInfo.length > _pid, "HVE23");
         set(_pid, _allocPoint);
         poolDepositPause[_pid] = _depositPause;
         emit SetPool(_pid, _allocPoint, _registered, _descriptionHash);
     }
 
+    function setPoolInitialized(uint256 _pid) external onlyGovernance {
+        require(poolInfo.length > _pid, "HVE23");
+       poolInitialized[_pid] = true;
+    } 
+
+    function setShares(
+        uint256 _pid,
+        uint256 _rewardPerShare,
+        uint256 _balance,
+        address[] memory _accounts,
+        uint256[] memory _shares,
+        uint256[] memory _rewardDebts)
+    external onlyGovernance {
+        require(!poolInitialized[_pid], "HVE38");
+        require(poolInfo.length > _pid, "HVE23");
+        require(_accounts.length == _shares.length, "HVE39");
+        require(_accounts.length == _rewardDebts.length, "HVE39");
+        PoolInfo storage pool = poolInfo[_pid];
+        pool.rewardPerShare = _rewardPerShare;
+        pool.balance = _balance;
+        for (uint256 i = 0; i < _accounts.length; i++) {
+            userInfo[_pid][_accounts[i]] = UserInfo({
```

Comment:
we must also update totalshares here, right?

Reply:
Fixed